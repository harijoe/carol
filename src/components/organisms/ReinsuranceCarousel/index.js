import React, { Component } from 'react'
import styled, { css } from 'styled-components'
import stripTags from 'utils/stripTags'
import { theme, breakpoint, mapBreakpoints } from 'utils/style'

import { Heading, Paragraph, Card } from 'components'
import { PostList } from 'containers'

const StyledAside = styled.aside`
  width: 100%;
  padding: 0;

  > div:first-child {
    display: flex;
    height: 100%;

    > div {
      max-width: 100%;
    }
  }

  .slick-dots {
    bottom: ${theme('spaces.xl')};
  }

  .slick-slider {
    margin: 0;
    padding: 0;
    margin-bottom: 5rem;

    &,
    .slick-list,
    .slick-track {
      display: block;
      height: 100%;
    }

    .slick-list {
      padding: 0;
    }
  }

  .slick-slide {
    float: left;
  }
`

const StyledItem = styled.div`
  ${({ src }) => `
  position: relative;
  display: flex;
  flex-direction: column;
  padding: ${theme('spaces.xxl')};
  height: 100%;
  background-image: url(${src});
  background-size: 100% 45rem;
  background-repeat:no-repeat;
  background-position:center top;
`};
`

const StyledHeading = styled(Heading)`
  position: relative;
  z-index: 10;
  width: 100%;
  font-size: ${theme('fonts.size.xl')};
  color: ${theme('colors.black')};
`

const StyledParagraph = styled(Paragraph)`
  position: relative;
  z-index: 10;
  width: 100%;
  line-height: 1.5;
  color: ${theme('colors.black')};
`

const StyledCard = styled(Card)`
  ${mapBreakpoints(
    bp => css`
    margin-left: ${theme(`grid.gutterWidth.${bp}`, 'rem')};
    margin-right: ${theme(`grid.gutterWidth.${bp}`, 'rem')};
  `,
  )}

  width: calc(100vw - 4.8rem);
  margin-bottom: 0.8rem;
  margin-top: 50%;
  padding: ${theme('spaces.xl')} ${theme('spaces.m')};

  ${breakpoint('m')`
    margin-left: auto;
    margin-right: auto;
    margin-top: calc(${theme('spaces.xxxl')} * 2.1);
    padding: ${theme('spaces.xxl')} ${theme('spaces.xxl')};
    min-height: 31.2rem;
    width: 48rem;
  `}
`

const WrapContent = styled.div`margin: 15rem ${theme('spaces.xl')} 0 ${theme('spaces.xl')};`

class ReinsuranceCarousel extends Component {
  state = {
    activeImage: 0,
  }

  afterChange = slideIndex => {
    this.setState({
      activeImage: slideIndex,
    })
  }

  generateChild = (i, { featuredMedia: { src }, title, body }) =>
    <StyledItem key={i} src={src}>
      <WrapContent>
        <StyledCard>
          <StyledHeading level={3}>
            {stripTags(title)}
          </StyledHeading>
          <StyledParagraph>
            {stripTags(body)}
          </StyledParagraph>
        </StyledCard>
      </WrapContent>
    </StyledItem>

  render() {
    return (
      <StyledAside>
        <PostList
          scope="reinsuranceArticles"
          tags={['api-quotatis-reinsurance']}
          limit={3}
          generateChild={this.generateChild}
          carousel={{
            infinite: true,
            afterChange: this.afterChange,
            slidesToShow: 1,
            slidesToScroll: 1,
            dots: true,
            autoplay: true,
            pauseOnHover: true,
            arrows: false,
          }}
        />
      </StyledAside>
    )
  }
}

export default ReinsuranceCarousel
