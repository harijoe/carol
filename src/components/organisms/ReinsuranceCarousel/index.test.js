import React from 'react'
import { shallow } from 'enzyme'

import ReinsuranceCarousel from './'

it('renders ReinsuranceCarousel', () => {
  expect(shallow(<ReinsuranceCarousel />)).toMatchSnapshot()
})
