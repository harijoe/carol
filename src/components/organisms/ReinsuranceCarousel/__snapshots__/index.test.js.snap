// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReinsuranceCarousel 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <StyledAside>
    <Mock
        carousel={
            Object {
                "afterChange": [Function],
                "arrows": false,
                "autoplay": true,
                "dots": true,
                "infinite": true,
                "pauseOnHover": true,
                "slidesToScroll": 1,
                "slidesToShow": 1,
              }
        }
        generateChild={[Function]}
        limit={3}
        scope="reinsuranceArticles"
        tags={
            Array [
                "api-quotatis-reinsurance",
              ]
        }
    />
</StyledAside>,
  "nodes": Array [
    <StyledAside>
      <Mock
            carousel={
                  Object {
                        "afterChange": [Function],
                        "arrows": false,
                        "autoplay": true,
                        "dots": true,
                        "infinite": true,
                        "pauseOnHover": true,
                        "slidesToScroll": 1,
                        "slidesToShow": 1,
                      }
            }
            generateChild={[Function]}
            limit={3}
            scope="reinsuranceArticles"
            tags={
                  Array [
                        "api-quotatis-reinsurance",
                      ]
            }
      />
</StyledAside>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ReinsuranceCarousel />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ReinsuranceCarousel {
        "_reactInternalInstance": [Circular],
        "afterChange": [Function],
        "context": Object {},
        "generateChild": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "activeImage": 0,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <StyledAside>
          <Mock
                    carousel={
                              Object {
                                        "afterChange": [Function],
                                        "arrows": false,
                                        "autoplay": true,
                                        "dots": true,
                                        "infinite": true,
                                        "pauseOnHover": true,
                                        "slidesToScroll": 1,
                                        "slidesToShow": 1,
                                      }
                    }
                    generateChild={[Function]}
                    limit={3}
                    scope="reinsuranceArticles"
                    tags={
                              Array [
                                        "api-quotatis-reinsurance",
                                      ]
                    }
          />
</StyledAside>,
        "_debugID": 2,
        "_renderedOutput": <StyledAside>
          <Mock
                    carousel={
                              Object {
                                        "afterChange": [Function],
                                        "arrows": false,
                                        "autoplay": true,
                                        "dots": true,
                                        "infinite": true,
                                        "pauseOnHover": true,
                                        "slidesToScroll": 1,
                                        "slidesToShow": 1,
                                      }
                    }
                    generateChild={[Function]}
                    limit={3}
                    scope="reinsuranceArticles"
                    tags={
                              Array [
                                        "api-quotatis-reinsurance",
                                      ]
                    }
          />
</StyledAside>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ReinsuranceCarousel />,
}
`;
